### DISCUSSION TITLE
Why doesn't FileManager have Ports?

### DISCUSSION THREAD

**Discussion by VR314** [nasa/fprime#2299] [Q&A] [answered]
I am curious why `Svc::FileManager` can only be interacted with using commands. Could you please clarify if there is a particular design principle or practical rationale behind the absence of any defined ports? If needed, I can make a PR to add ports to the `Svc::FileManager` component consistent with the command definitions.

**Comment by timcanham** [comment, marked as answer]
`Svc::FileManager` does have ports, but only the standard F Prime ports:

https://github.com/nasa/fprime/blob/devel/Svc/FileManager/FileManager.fpp

Other ports are added if a component needs to interact with other components other than the "standard" F Prime infrastructure components. So far, we haven't identified a need for `Svc::FileManager` to do that.
